mimic/rest/auth_api.py

```diff
@@ -88,35 +90,33 @@ def lookup(entry):
                                   exception.session.username,
                                   exception.session.user_id))
 
-        if content['auth'].get('passwordCredentials'):
-            username = content['auth']['passwordCredentials']['username']
-            password = content['auth']['passwordCredentials']['password']
-            return format_response(
-                lambda: self.core.sessions.session_for_username_password(
-                    username, password, tenant_id),
-                username_generator)
+        cred_mappings = {
```

Comment:
Just a thought for future PRs: this mapping here is really populated with keys that are mostly within the domain of the credentials themselves, i.e. `RAX-KSKEY:apiKeyCredentials` is a string that really only `APIKeyCredentials` ought to know about.  So this should really be constructing this dictionary by looping over `PasswordCredentials`/`APIKeyCredentials`/`TokenCredentials`… and once it's abstracted out like that, you could even have a plugin API :).  (Which might be useful given the Kerberos and OAuth stuff I keep hearing rumours about)